app/cloud.go

```diff
@@ -156,3 +157,30 @@ func (a *App) SendCloudTrialEndedEmail() *model.AppError {
 
 	return nil
 }
+
+// GetCloudUsageForMessages returns posts usage percentage against total available limit on Cloud.
+// Usage percentage is returned in multiples of 10 eg. 0, 10, 20
+func (a *App) GetCloudUsageForMessages(userID string) (int, *model.AppError) {
+	// Fetch limit
+	limits, err := a.Cloud().GetCloudLimits(userID)
+	if err != nil {
+		return 0, model.NewAppError("GetCloudUsageForMessages", "app.channel.GetCloudLimits.internal_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Fetch messages count
+	c, err := a.Srv().Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true})
+	if err != nil {
+		return 0, model.NewAppError("GetCloudUsageForMessages", "app.channel.AnalyticsPostCount.internal_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Return usage %
+	max := float64(*limits.Messages.History)
+	count := float64(c)
+
+	if count >= max {
+		return 100, nil
+	}
+
+	usage := (count / max) * 100
+	return utils.FloorToNearest10(usage), nil
```

Comment:
As per this change, I don't need to call the getCloudLimits at all. I could just return the roundOff count of posts, right?
And that would mean, it could be just a regular posts/ api instead of anything to do with cloud.
Or should I just remove the getCloudLimits call and keep the rest files/api same?

Reply:
I'm 3/5 to have a `/limits` path. Messages and storage both could fall under this path.
@Willyfrog @ashishbhate thoughts?

Reply:
Yeah, I think a `/limits` subpath is the way to go. 

It appears Mattermost routes are grouped up by what is being acted on, returned, or contained within a sub-router.

I think it makes sense for `limits/posts` to return the posts limit because we're returning limits, posts just happen to be one of the limits. 

`posts/limit` doesn't seem right because we aren't acting on posts, returning posts and limits aren't contained within a post.

Reply:
This is bike-shedding but I think the subpath should probably be `/usage` instead of `limits`. I'm 0/5.

Reply:
* Alright, lets finalize it to `/usage/posts`. 
* And I'll remove the cloud call as well and just return the posts count. 
* That would also mean, no `enabled` field, just a `count` field will do.

Although, now posts count raise another question like up to what level do we want to round off?
* The current method rounds off only to nearest 10, i.e., 9876 to 9870.
  * Will this work? 
* Or should I make it round off to the last possible digit like 9876 to 9000?

Reply:
>(regular user view https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=602%3A430680)


@neallred This mock up is under the "admin" view (Frame 957). 

From the regular user views (Frame  958), it looks like a regular user doesn't need access  to this API. The limits check can happen transparently in the app. 

Can we continue limiting access to this API to sysadmin users only?

Reply:
There's a number of cases where end users need the usage data, too, so I think anyone with a session needs to be able to access.

The usage & limits modal shows up in a number of non-admin user contexts, too. For example, in boards: https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=784%3A453152 , and I think in channels, too, but I need to check more.

The channels main menu also sometimes needs usage data for end users: https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=121%3A333428

Reply:
> Although, now posts count raise another question like up to what level do we want to round off?

No strong opinion from me. The UI currently only shows in increments of 100, e.g. `5.5K messages`, so no benefit returning anything more granular.

Reply:
@neallred I'll follow suit with a `/usage/integrations` endpoint